arches_references/src/arches-references/components/tree/ListTreeControls.vue

```diff
@@ -20,9 +31,75 @@ const isMultiSelecting = defineModel<boolean>("isMultiSelecting", {
     required: true,
 });
 const nextNewList = defineModel<NewControlledList>("nextNewList");
-const newListFormValue = defineModel<string>("newListFormValue", {
-    required: true,
-});
+const { newListName } = defineProps<{ newListName: string }>();
+const { setDisplayedRow } = inject(
+    displayedRowKey,
+) as DisplayedListItemRefAndSetter;
+const route = useRoute();
+watch(
```

Comment:
what's going on in this file? Also I don't _love_ the idea of subcomponents being able to navigate, is it possible to abstract this to the main component?

Reply:
I moved this up one component to `<ListTree>` where the rest of the list tree state is owned.

What's going on is that the `<ListTree>` needs to react to route changes:
- tree rows need to expand and contract
- tree rows need to be selected

> Also I don't love the idea of subcomponents being able to navigate, is it possible to abstract this to the main component?

I guess I wouldn't say that the subcomponent is navigating: it's just reacting and manipulating its own state in response to route changes. I gave it a little thought, and I don't think it makes sense to put the tree expanded and selected keys state on the top-level component. I'd rather adopt a Redux/Pinia store pattern before doing that. The client state management would be a little more compact if we were using Pinia.

You'll notice the main component is in fact where `router.push()` happens.

Let me know if I can clarify this with variable naming or comments, or if you have other suggestions.